packages/block-library/src/navigation/style.scss

```diff
@@ -488,6 +488,13 @@ button.wp-block-navigation-item__content {
 		align-items: var(--navigation-layout-align, initial);
 	}
 
+	// If the responsive wrapper is present but overlay is not open, overlay colors
+	// shouldn't apply and !important is to override color classes or inline styles.
+	&:not(.is-menu-open) {
+		color: inherit !important;
+		background-color: inherit !important;
+	}
+
```

Comment:
I had removed these because these properties were set to `inherit` and not realizing the `!important` bit was key. I double-checked and found that out. Removing them hadn’t caused the e2e tests for colors to fail so along with restoring this I added an e2e test that fails without these.